features/login/impl/src/main/kotlin/io/element/android/features/login/impl/root/LoginRootState.kt

```diff
@@ -27,8 +27,8 @@ data class LoginRootState(
     val formState: LoginFormState,
     val eventSink: (LoginRootEvents) -> Unit
 ) {
-    val submitEnabled =
-        formState.login.isNotEmpty() && formState.password.isNotEmpty() && loggedInState != LoggedInState.LoggingIn
+    val submitEnabled: Boolean get() =
+        formState.login.isNotEmpty() && formState.password.isNotEmpty() && loggedInState !is LoggedInState.ErrorLoggingIn
```

Comment:
In case of error, the button has to be enabled, no? To let the user retry? So maybe simplify with:

```suggestion
        formState.login.isNotEmpty() && formState.password.isNotEmpty()
```

Reply:
Not really, when the error is received an alert dialog is displayed, and that error is cleared as soon as the dialog disappears. Technically, it's not needed to disabled the button since the alert will be blocking the UI, but it seemed to be clearer that way that you couldn't advance until the error was accepted.